test/error_messages.coffee

```diff
@@ -1742,3 +1742,28 @@ test "#3199: error message for throw indented comprehension", ->
       x for x in [1, 2, 3]
       ^
   '''
+
+test "#2998: Indentation mismatch in object shouldn’t produce two objects", ->
+  assertErrorFormat '''
+  colors =
+     45: 'cyan'
+     52: 'sepia'
+    100: 'olive'
+    124: 'red'
+  ''', '''
+    [stdin]:4:3: error: unexpected indentation
+      100: 'olive'
+      ^
+  '''
+
+  # Former object test that is now invalid because it looks too much like a mistake.
+  # It used to test that string-keyed objects shouldn’t suppress newlines.
+  assertErrorFormat '''
+    one =
+      '>!': 3
+    six: -> 10
```

Comment:
What I mean is:

```coffee
a =        # indent: 0 - block starts at indent 0
  '>!': 3  # indent: 2 - sets the indentation for everything in the objects
 bad: 1    # indent: 1 - invalid, everything in the objects needs to be indent 2
six: -> 10 # indent: 0 - new expression

foo = ->
  a =        # indent: 2 - block starts at indent 2
    '>!': 3  # indent: 4 - sets the indentation for everything in the objects
   bad: 1    # indent: 3 - invalid, everything in the objects needs to be indent 4
  six: -> 10 # indent: 2 - new expression
```

Here's a contrived example that uses it in a visible way:

```coffee
o = null
f = ->
  o = 
    '>!': 3
  six: -> 10

result = f() # { six: -> 10 }
o            # { '>!': 3 }
```

But my main issue is that it should be allowed for the same reason the following are:

```coffee
1
'hello'
-> foo
etc
```